src/context.tsx

```diff
@@ -190,22 +176,27 @@ const reducer = (state: State, action: Record<string, any>): State => {
       const pageState = newState[pageStateEnum as keyof State] as PageState;
       pageState.isLoading = isLoading;
       return newState;
-    };
+    }
     case "UPDATE_AGGREGATION_FACTOR":
       return { ...state, chartAggregationFactor: action.payload };
     case "UPDATE_EXPLORE_IS_OPEN":
```

Comment:
Could be wise to address in another PR but this whole system around our `dataPageState` object is very outdated. We should update this logic to be simpler and tie up loose ends eventually. Maybe we have a quick sync about this sometime @brett-stephen @annoyingpr

Reply:
Agreed. Also, a bit unrelated but it would also be nice to address the issue of the API calls saved in the app context. If you recall, we didn't have a way to distinguish between a failed API call and one that hasn't returned anything yet.

Reply:
Sounds good, do you think it should be a part of this [ticket](https://airtable.com/tbli2lWQHAqBa6ZcI/viw8Ro4zYPYcBGXRw/recNqmd8naERYKBio?blocks=hide) or should we create a new one?

Reply:
make new one pls